libkpod/containerserver.go

```diff
@@ -51,12 +55,50 @@ func (c *ContainerServer) ImageContext() *types.SystemContext {
 
 // New creates a new ContainerServer
 func New(runtime *oci.Runtime, store cstorage.Store, storageImageServer storage.ImageServer, ctrNameIndex *registrar.Registrar, ctrIDIndex *truncindex.TruncIndex, imageContext *types.SystemContext) *ContainerServer {
+	containers := oci.NewMemoryStore()
 	return &ContainerServer{
 		runtime:            runtime,
 		store:              store,
 		storageImageServer: storageImageServer,
 		ctrNameIndex:       ctrNameIndex,
 		ctrIDIndex:         ctrIDIndex,
 		imageContext:       imageContext,
+		stateLock:          new(sync.Mutex),
+		state: &containerServerState{
+			containers: containers,
+		},
 	}
 }
+
+type containerServerState struct {
+	containers oci.ContainerStorer
+}
+
+// AddContainer adds a container to the container state store
+func (c *ContainerServer) AddContainer(ctr *oci.Container) {
+	c.stateLock.Lock()
+	c.state.containers.Add(ctr.ID(), ctr)
+	c.stateLock.Unlock()
+}
+
+// GetContainer returns a container by its ID
+func (c *ContainerServer) GetContainer(id string) *oci.Container {
+	c.stateLock.Lock()
+	ctr := c.state.containers.Get(id)
+	c.stateLock.Unlock()
+	return ctr
```

Comment:
I think these would look better as
```
	c.stateLock.Lock()
        defer c.stateLock.Unlock()
	return c.state.containers.Get(id)
```
I think you should change all of them to look this way. cleaner and shorter, and less likely for someone at some point in the future to add code and forget to unlock.